index.js

```diff
@@ -1,88 +1,118 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
-
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+let counter = 1 
+counter = 2
+console.log("%cQuestion 1", "color: red") 
 
+console.log(counter) 
+// => 2
+console.log("%c----------", "color: red") 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+console.log("%cQuestion 2", "color: red") 
+
+const name = "Raffy"
 
-// console.log(name)
+console.log(name)
 // => "Raffy"
 // name = "Not Raffy" 
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
+function drinkWater(currentThirstLevel) {
+  console.log("Man I sure am thirsty")
+  currentThirstLevel -= 1 
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
+
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum) { 
+  const maybeNumInt = parseInt(maybeNum, 10)
+  if (Number.isInteger(num) && num === maybeNum)
+    return "same same"
+  if (num === maybeNumInt)
```

Comment:
The `==` operator coerces the `maybeNumInt` into the same dataType as num!
But, I like what you're doing here - being explicit with the conversion and then using the `===`